application/CohortManager/src/Functions/CaasIntegration/receiveCaasFile/ProcessFileClasses/ProcessCaasFile.cs

```
@@ -145,18 +150,36 @@ private async Task AddBatchToQueue(Batch currentBatch, string name)
                 await RemoveParticipant(updateRecords, name);
             }
         }
+        // this used to release memory from being used 
+        currentBatch = null;
     }
 
     private async Task UpdateParticipant(BasicParticipantCsvRecord basicParticipantCsvRecord, string name)
     {
         try
         {
             var json = JsonSerializer.Serialize(basicParticipantCsvRecord);
-            if (await _checkDemographic.PostDemographicDataAsync(basicParticipantCsvRecord.participant, Environment.GetEnvironmentVariable("DemographicURI")))
+            var listOfData = new List<ParticipantDemographic>()
+            {
+                basicParticipantCsvRecord.participant.ToParticipantDemographic()
+            };
+
+            var participantRecord = await _participantDemographic.GetByFilter(x => x.NhsNumber.ToString() == basicParticipantCsvRecord.participant.NhsNumber);
+
+            var participant = participantRecord.FirstOrDefault();
+            if (participant != null)
+            {
+                await _participantDemographic.Delete(participant.ParticipantId.ToString());
+                if (await _checkDemographic.PostDemographicDataAsync(listOfData, Environment.GetEnvironmentVariable("DemographicURI") ?? ""))
```

I don't think we should be null coalescing here is a good idea.
if we don't have the env we should just throw an exception and log it rather than trying to call the api?